cmake/FindFuthark.cmake

```diff
@@ -0,0 +1,125 @@
+# TRACCC library, part of the ACTS project (R&D line)
+#
+# (c) 2021-2022 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# The C language is not enabled in our project by default, but we need it to
+# compile Futhark's code.
+enable_language(C)
+
+# Try to find our dependencies, which are of course Futhark itself (required)
+# and the CUDA toolkit for compiling CUDA projects. Also notify the user of
+# what we find.
+find_program(Futhark futhark REQUIRED)
+find_package(CUDAToolkit)
```

Comment:
Later in the code, before you try to make use of the targets/variables set by this find-module, you should check whether `CUDAToolkit_FOUND` is `TRUE` or not. And give a meaningful error if the user asked for CUDA sources without CUDA being available.